src/Umbraco.Community.AzureSSO/MicrosoftAccountAuthenticationExtensions.cs

```diff
@@ -16,31 +18,61 @@ internal static IUmbracoBuilder AddMicrosoftAccountAuthentication(this IUmbracoB
 			var azureSsoConfiguration = new AzureSSOConfiguration();
 			builder.Config.Bind(AzureSSOConfiguration.AzureSsoSectionName, azureSsoConfiguration);
 
-			builder.Services.AddSingleton<AzureSsoSettings>(conf => new AzureSsoSettings(azureSsoConfiguration));
+			var settings = new AzureSsoSettings(azureSsoConfiguration);
+			builder.Services.AddSingleton<AzureSsoSettings>(conf => settings);
 			builder.Services.ConfigureOptions<MicrosoftAccountBackOfficeExternalLoginProviderOptions>();
 			
 			var initialScopes = Array.Empty<string>();
 			builder.AddBackOfficeExternalLogins(logins =>
-			{
-				logins.AddBackOfficeLogin(
-					backOfficeAuthenticationBuilder =>
+				{
+					foreach (var profile in settings.Profiles)
 					{
-						backOfficeAuthenticationBuilder.AddMicrosoftIdentityWebApp(options =>
-								{
-									builder.Config.Bind(AzureSSOConfiguration.AzureSsoCredentialSectionName, options);
-									options.SignInScheme = backOfficeAuthenticationBuilder.SchemeForBackOffice(MicrosoftAccountBackOfficeExternalLoginProviderOptions.SchemeName);
-									options.Events = new OpenIdConnectEvents();
-								},
-								options => { builder.Config.Bind(AzureSSOConfiguration.AzureSsoCredentialSectionName, options); },
-								displayName: azureSsoConfiguration.DisplayName ?? "Azure Active Directory",
-								openIdConnectScheme: backOfficeAuthenticationBuilder.SchemeForBackOffice(MicrosoftAccountBackOfficeExternalLoginProviderOptions.SchemeName) ?? String.Empty)
-							.EnableTokenAcquisitionToCallDownstreamApi(options => builder.Config.Bind(AzureSSOConfiguration.AzureSsoCredentialSectionName, options), initialScopes)
-							.AddTokenCaches(azureSsoConfiguration.TokenCacheType);
-					});
-			});
+						logins.AddBackOfficeLogin(
+							backOfficeAuthenticationBuilder =>
+							{
+								backOfficeAuthenticationBuilder.AddMicrosoftIdentityWebApp(options =>
+										{
+											CopyCredentials(options, profile.Credentials);
+											options.SignInScheme = backOfficeAuthenticationBuilder.SchemeForBackOffice(profile.Name);
+											options.Events = new OpenIdConnectEvents();
+											
+										},
+										displayName: profile.DisplayName ?? "Azure Active Directory",
```

Comment:
Should we keep up with Microsoft's (silly) renaming of AAD to "Microsoft Entra ID"?